File: suc/kotlin/087.kt
package com.example.myapplication

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Check
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.PlayArrow
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Configuration for Immersive Mode (Hide System Bars)
        val windowInsetsController = WindowCompat.getInsetsController(window, window.decorView)
        windowInsetsController.systemBarsBehavior =
            WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        windowInsetsController.hide(WindowInsetsCompat.Type.systemBars())

        setContent {
            // Apply a material theme wrapper, but force light colors for the requirement
            MaterialTheme(
                colorScheme = androidx.compose.material3.lightColorScheme(
                    background = Color.White,
                    surface = Color.White,
                    primary = Color(0xFF6D4C41), // Brownish tea color
                    secondary = Color(0xFFA1887F),
                    onPrimary = Color.White
                )
            ) {
                TeaBrewingScreen()
            }
        }
    }
}

@Composable
fun TeaBrewingScreen() {
    Surface(
        modifier = Modifier.fillMaxSize(),
        color = Color.White // Global White Background Constraint
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(24.dp),
            verticalArrangement = Arrangement.Top,
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // Top Header
            HeaderSection()

            Spacer(modifier = Modifier.height(32.dp))

            // Main Card displaying the selected Tea
            TeaSelectionCard()

            Spacer(modifier = Modifier.height(32.dp))

            // Timer Visualization
            TimerDisplay(time = "02:00")

            Spacer(modifier = Modifier.height(40.dp))

            // Action Buttons
            ControlButtons()

            Spacer(modifier = Modifier.weight(1f))

            // Preparation Steps
            PreparationSteps()
        }
    }
}

@Composable
fun HeaderSection() {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Column {
            Text(
                text = "Good Morning,",
                style = MaterialTheme.typography.bodyMedium,
                color = Color.Gray
            )
            Text(
                text = "Tea Master",
                style = MaterialTheme.typography.titleLarge,
                fontWeight = FontWeight.Bold,
                color = Color.Black
            )
        }
        IconButton(
            onClick = { },
            modifier = Modifier
                .size(48.dp)
                .background(Color(0xFFF5F5F5), CircleShape)
        ) {
            Icon(
                imageVector = Icons.Default.Settings,
                contentDescription = "Settings",
                tint = Color.Black
            )
        }
    }
}

@Composable
fun TeaSelectionCard() {
    Card(
        colors = CardDefaults.cardColors(containerColor = Color(0xFFFAFAFA)),
        border = BorderStroke(1.dp, Color(0xFFEEEEEE)),
        shape = RoundedCornerShape(24.dp),
        modifier = Modifier.fillMaxWidth()
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(20.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Abstract Tea Icon drawn with Canvas
            Box(
                modifier = Modifier
                    .size(60.dp)
                    .background(Color(0xFFEFEBE9), CircleShape),
                contentAlignment = Alignment.Center
            ) {
                Canvas(modifier = Modifier.size(30.dp)) {
                    // Draw a simple leaf shape
                    drawCircle(
                        color = Color(0xFF6D4C41),
                        radius = size.minDimension / 2,
                        style = Stroke(width = 2.dp.toPx())
                    )
                    drawLine(
                        color = Color(0xFF6D4C41),
                        start = Offset(size.width / 2, size.height / 4),
                        end = Offset(size.width / 2, size.height * 0.75f),
                        strokeWidth = 2.dp.toPx(),
                        cap = StrokeCap.Round
                    )
                }
            }

            Spacer(modifier = Modifier.width(16.dp))

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = "Earl Grey Reserve",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black
                )
                Text(
                    text = "Black Tea • 95°C",
                    style = MaterialTheme.typography.bodySmall,
                    color = Color(0xFF8D6E63)
                )
            }

            Icon(
                imageVector = Icons.Default.Favorite,
                contentDescription = "Favorite",
                tint = Color(0xFFFFCDD2) // Light red tint
            )
        }
    }
}

@Composable
fun TimerDisplay(time: String) {
    Box(
        contentAlignment = Alignment.Center,
        modifier = Modifier.size(220.dp)
    ) {
        // Outer Progress Ring
        Canvas(modifier = Modifier.fillMaxSize()) {
            drawCircle(
                color = Color(0xFFF5F5F5),
                style = Stroke(width = 12.dp.toPx())
            )
            drawArc(
                color = Color(0xFF6D4C41),
                startAngle = -90f,
                sweepAngle = 240f,
                useCenter = false,
                style = Stroke(width = 12.dp.toPx(), cap = StrokeCap.Round)
            )
        }

        // Inner Content
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(
                text = "Steeping",
                style = MaterialTheme.typography.labelMedium,
                color = Color.Gray,
                letterSpacing = 2.sp
            )
            Text(
                text = time,
                style = MaterialTheme.typography.displayMedium,
                fontWeight = FontWeight.Light,
                color = Color.Black
            )
        }
    }
}

@Composable
fun ControlButtons() {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.Center,
        verticalAlignment = Alignment.CenterVertically
    ) {
        // Reset Button
        OutlinedButton(
            onClick = { },
            shape = CircleShape,
            modifier = Modifier.size(60.dp),
            border = BorderStroke(1.dp, Color(0xFFE0E0E0)),
            contentPadding = androidx.compose.foundation.layout.PaddingValues(0.dp)
        ) {
            Icon(
                imageVector = Icons.Default.Refresh,
                contentDescription = "Reset",
                tint = Color.Gray
            )
        }

        Spacer(modifier = Modifier.width(24.dp))

        // Play/Pause Button
        Button(
            onClick = { },
            shape = RoundedCornerShape(24.dp),
            modifier = Modifier
                .height(60.dp)
                .width(120.dp),
            colors = ButtonDefaults.buttonColors(
                containerColor = Color(0xFF6D4C41)
            )
        ) {
            Icon(
                imageVector = Icons.Default.PlayArrow,
                contentDescription = "Start",
                modifier = Modifier.padding(end = 8.dp)
            )
            Text(text = "Start")
        }

        Spacer(modifier = Modifier.width(24.dp))

        // Add Note Button
        OutlinedButton(
            onClick = { },
            shape = CircleShape,
            modifier = Modifier.size(60.dp),
            border = BorderStroke(1.dp, Color(0xFFE0E0E0)),
            contentPadding = androidx.compose.foundation.layout.PaddingValues(0.dp)
        ) {
            Icon(
                imageVector = Icons.Default.Add,
                contentDescription = "Add Note",
                tint = Color.Gray
            )
        }
    }
}

@Composable
fun PreparationSteps() {
    Column(modifier = Modifier.fillMaxWidth()) {
        Text(
            text = "PREPARATION",
            style = MaterialTheme.typography.labelSmall,
            fontWeight = FontWeight.Bold,
            color = Color.LightGray,
            modifier = Modifier.padding(bottom = 12.dp)
        )

        StepItem(step = "1", text = "Boil fresh water to 95°C")
        StepItem(step = "2", text = "Add 1 tsp of loose leaf per cup")
        StepItem(step = "3", text = "Pour water and wait for timer")
    }
}

@Composable
fun StepItem(step: String, text: String) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Box(
            modifier = Modifier
                .size(24.dp)
                .background(Color(0xFFF5F5F5), CircleShape),
            contentAlignment = Alignment.Center
        ) {
            Text(
                text = step,
                style = MaterialTheme.typography.labelSmall,
                color = Color.Black,
                fontWeight = FontWeight.Bold
            )
        }
        Spacer(modifier = Modifier.width(12.dp))
        Text(
            text = text,
            style = MaterialTheme.typography.bodyMedium,
            color = Color.DarkGray
        )
        Spacer(modifier = Modifier.weight(1f))
        Icon(
            imageVector = Icons.Default.Check,
            contentDescription = null,
            tint = Color(0xFFEEEEEE),
            modifier = Modifier.size(16.dp)
        )
    }
    HorizontalDivider(color = Color(0xFFFAFAFA))
}

@Preview(showBackground = true, widthDp = 360, heightDp = 800)
@Composable
fun PreviewTeaApp() {
    MaterialTheme(
        colorScheme = androidx.compose.material3.lightColorScheme(
            background = Color.White,
            surface = Color.White
        )
    ) {
        TeaBrewingScreen()
    }
}

File: suc/swift/087.swift
import SwiftUI

@main
struct TeaMasterApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

struct ContentView: View {
    var body: some View {
        TeaBrewingScreen()
            .preferredColorScheme(.light)
            .statusBarHidden(true)
    }
}

// MARK: - App Colors
struct TeaColors {
    static let primary = Color(red: 0.43, green: 0.30, blue: 0.25) // #6D4C41
    static let secondary = Color(red: 0.63, green: 0.53, blue: 0.50) // #A1887F
    static let surfaceVariant = Color(red: 0.98, green: 0.98, blue: 0.98) // #FAFAFA
    static let outline = Color(red: 0.93, green: 0.93, blue: 0.93) // #EEEEEE
    static let backgroundTint = Color(red: 0.96, green: 0.96, blue: 0.96) // #F5F5F5
    static let teaLeafDetail = Color(red: 0.55, green: 0.43, blue: 0.39) // #8D6E63
    static let favoriteTint = Color(red: 1.0, green: 0.80, blue: 0.82) // #FFCDD2
    static let textSecondary = Color.gray
    static let textPrimary = Color.black
}

// MARK: - Main Screen
struct TeaBrewingScreen: View {
    var body: some View {
        ZStack {
            Color.white.ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Top Header
                HeaderSection()
                    .padding(.top, 24)
                    .padding(.bottom, 32)
                
                // Main Card
                TeaSelectionCard()
                    .padding(.bottom, 32)
                
                // Timer
                TimerDisplay(time: "02:00")
                    .padding(.bottom, 40)
                
                // Action Buttons
                ControlButtons()
                
                Spacer()
                
                // Steps
                PreparationSteps()
                    .padding(.bottom, 24)
            }
            .padding(.horizontal, 24)
        }
    }
}

// MARK: - Components

struct HeaderSection: View {
    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: 2) {
                Text("Good Morning,")
                    .font(.system(size: 14)) // BodyMedium approx
                    .foregroundColor(TeaColors.textSecondary)
                Text("Tea Master")
                    .font(.system(size: 22, weight: .bold)) // TitleLarge approx
                    .foregroundColor(TeaColors.textPrimary)
            }
            Spacer()
            Button(action: {}) {
                Image(systemName: "gearshape.fill")
                    .font(.system(size: 24))
                    .foregroundColor(TeaColors.textPrimary)
                    .frame(width: 48, height: 48)
                    .background(TeaColors.backgroundTint)
                    .clipShape(Circle())
            }
        }
    }
}

struct TeaSelectionCard: View {
    var body: some View {
        HStack(spacing: 16) {
            // Icon Box
            ZStack {
                Circle()
                    .fill(Color(red: 0.94, green: 0.92, blue: 0.91)) // #EFEBE9
                    .frame(width: 60, height: 60)
                
                // Custom drawn leaf-like shape using SF Symbol or Path
                // Approximating the canvas drawing with a symbol + styling
                Image(systemName: "leaf") // Simple fallback for leaf concept
                    .resizable()
                    .aspectRatio(contentMode: .fit)
                    .frame(width: 24, height: 24)
                    .foregroundColor(TeaColors.primary)
                    // Or custom drawing to match exactly:
                    // But requirement says "maintain component styles".
                    // Let's use a custom shape to mimic the Kotlin Canvas drawing.
                    .opacity(0) // Hiding SF Symbol to draw custom
                    .overlay(
                        CustomLeafIcon()
                            .frame(width: 30, height: 30)
                    )
            }
            
            VStack(alignment: .leading, spacing: 2) {
                Text("Earl Grey Reserve")
                    .font(.system(size: 16, weight: .bold)) // TitleMedium approx
                    .foregroundColor(TeaColors.textPrimary)
                Text("Black Tea • 95°C")
                    .font(.system(size: 12)) // BodySmall approx
                    .foregroundColor(TeaColors.teaLeafDetail)
            }
            
            Spacer()
            
            Image(systemName: "heart.fill")
                .font(.system(size: 24))
                .foregroundColor(TeaColors.favoriteTint)
        }
        .padding(20)
        .background(TeaColors.surfaceVariant)
        .cornerRadius(24)
        .overlay(
            RoundedRectangle(cornerRadius: 24)
                .stroke(TeaColors.outline, lineWidth: 1)
        )
    }
}

struct CustomLeafIcon: View {
    var body: some View {
        GeometryReader { geometry in
            let w = geometry.size.width
            let h = geometry.size.height
            
            ZStack {
                // Circle outline
                Circle()
                    .stroke(TeaColors.primary, lineWidth: 2)
                    .frame(width: min(w, h), height: min(w, h))
                
                // Vertical Line
                Path { path in
                    path.move(to: CGPoint(x: w / 2, y: h / 4))
                    path.addLine(to: CGPoint(x: w / 2, y: h * 0.75))
                }
                .stroke(TeaColors.primary, style: StrokeStyle(lineWidth: 2, lineCap: .round))
            }
        }
    }
}

struct TimerDisplay: View {
    let time: String
    
    var body: some View {
        ZStack {
            // Background Circle
            Circle()
                .stroke(TeaColors.backgroundTint, lineWidth: 12)
                .frame(width: 220, height: 220)
            
            // Progress Arc
            Circle()
                .trim(from: 0.0, to: 0.66) // 240 degrees / 360
                .stroke(TeaColors.primary, style: StrokeStyle(lineWidth: 12, lineCap: .round))
                .rotationEffect(.degrees(-210)) // Start from bottom left-ish to match visual
                .frame(width: 220, height: 220)
            
            VStack(spacing: 4) {
                Text("Steeping")
                    .font(.system(size: 12, weight: .medium)) // LabelMedium
                    .tracking(2)
                    .foregroundColor(TeaColors.textSecondary)
                
                Text(time)
                    .font(.system(size: 45, weight: .light)) // DisplayMedium approx
                    .foregroundColor(TeaColors.textPrimary)
            }
        }
    }
}

struct ControlButtons: View {
    var body: some View {
        HStack(spacing: 24) {
            // Reset
            Button(action: {}) {
                Image(systemName: "arrow.counterclockwise")
                    .font(.system(size: 24))
                    .foregroundColor(.gray)
                    .frame(width: 60, height: 60)
                    .background(Color.white)
                    .clipShape(Circle())
                    .overlay(
                        Circle().stroke(Color(red: 0.88, green: 0.88, blue: 0.88), lineWidth: 1)
                    )
            }
            
            // Start
            Button(action: {}) {
                HStack(spacing: 8) {
                    Image(systemName: "play.fill")
                        .font(.system(size: 18))
                    Text("Start")
                        .font(.system(size: 16, weight: .medium))
                }
                .foregroundColor(.white)
                .frame(width: 120, height: 60)
                .background(TeaColors.primary)
                .cornerRadius(24)
            }
            
            // Add Note
            Button(action: {}) {
                Image(systemName: "plus")
                    .font(.system(size: 24))
                    .foregroundColor(.gray)
                    .frame(width: 60, height: 60)
                    .background(Color.white)
                    .clipShape(Circle())
                    .overlay(
                        Circle().stroke(Color(red: 0.88, green: 0.88, blue: 0.88), lineWidth: 1)
                    )
            }
        }
    }
}

struct PreparationSteps: View {
    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            Text("PREPARATION")
                .font(.system(size: 11, weight: .bold)) // LabelSmall
                .foregroundColor(.gray.opacity(0.8))
                .padding(.bottom, 12)
            
            StepItem(step: "1", text: "Boil fresh water to 95°C")
            Divider().background(TeaColors.backgroundTint)
            StepItem(step: "2", text: "Add 1 tsp of loose leaf per cup")
            Divider().background(TeaColors.backgroundTint)
            StepItem(step: "3", text: "Pour water and wait for timer")
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

struct StepItem: View {
    let step: String
    let text: String
    
    var body: some View {
        HStack(alignment: .center, spacing: 12) {
            Text(step)
                .font(.system(size: 11, weight: .bold))
                .foregroundColor(TeaColors.textPrimary)
                .frame(width: 24, height: 24)
                .background(TeaColors.backgroundTint)
                .clipShape(Circle())
            
            Text(text)
                .font(.system(size: 14)) // BodyMedium
                .foregroundColor(Color.gray)
            
            Spacer()
            
            Image(systemName: "checkmark")
                .font(.system(size: 12))
                .foregroundColor(TeaColors.outline)
        }
        .padding(.vertical, 8)
    }
}